packages/grafana-ui/src/components/Tags/Tag.tsx

```diff
@@ -47,8 +48,26 @@ export const Tag = forwardRef<HTMLElement, Props>(({ name, onClick, icon, classN
     </span>
   );
 });
+TagComponent.displayName = 'Tag';
 
-Tag.displayName = 'Tag';
+const TagSkeleton = () => {
+  const styles = useStyles2(getSkeletonStyles);
+  return <Skeleton width={60} height={22} containerClassName={styles.container} />;
+};
+
+interface TagWithSkeleton extends React.ForwardRefExoticComponent<Props & React.RefAttributes<HTMLElement>> {
+  Skeleton: typeof TagSkeleton;
+}
+
+export const Tag: TagWithSkeleton = Object.assign(TagComponent, {
+  Skeleton: TagSkeleton,
+});
+
```

Comment:
Can we do something similar to what Card does?

https://github.com/grafana/grafana/blob/c63a3796ffee9013f037d26e15465b7aecab9f7c/packages/grafana-ui/src/components/Card/Card.tsx/#L28-L36

Reply:
@joshhunt i don't understand, that's what it's doing already isn't it? the type it extends is just slightly more complicated because the component is forwarding a ref 🤔